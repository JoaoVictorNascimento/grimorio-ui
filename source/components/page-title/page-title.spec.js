import '../../../internals/test/helper';

import PageTitle from './page-title-component';

/** @test {PageTitle} */
describe('PageTitle component', () => {
  const emptyProps = {
    title: 'tÃ­tulo do header',
    sideComponent: undefined,
  }

  let wrapper

  beforeAll(() => {
    wrapper = shallow(
      <PageTitle {...emptyProps} />
    );
  })

  describe('#render', () => {
    it('render default', () => {
      expect(wrapper.debug()).toMatchSnapshot();
    });

    it('render with side content', () => {
      wrapper.setProps({...emptyProps, sideComponent: <div>Elemento da lateral</div>})
      expect(wrapper.debug()).toMatchSnapshot();
    })
  });
});
