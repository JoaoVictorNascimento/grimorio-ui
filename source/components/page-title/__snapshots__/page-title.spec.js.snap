// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle component #render render default 1`] = `
"<Panel size=\\"small\\" contentClassName=\\"titleWrap\\" title={false} loading={false} accordion={false} open={true} highlight=\\"shadow\\">
  <h1 className=\\"title\\">
    título do header
  </h1>
</Panel>"
`;

exports[`PageTitle component #render render with side content 1`] = `
"<Panel size=\\"small\\" contentClassName=\\"titleWrap\\" title={false} loading={false} accordion={false} open={true} highlight=\\"shadow\\">
  <h1 className=\\"title\\">
    título do header
  </h1>
  <div>
    <div>
      Elemento da lateral
    </div>
  </div>
</Panel>"
`;
